nomad/state/state_store.go

```diff
@@ -3442,6 +3442,21 @@ func (s *StateStore) updateDeploymentWithAlloc(index uint64, alloc, existing *st
 	state.HealthyAllocs += healthy
 	state.UnhealthyAllocs += unhealthy
 
+	// Ensure PlacedCanaries accurately reflects the alloc canary status
+	if alloc.DeploymentStatus != nil {
+		if alloc.DeploymentStatus.Canary {
+			found := false
+			for _, canary := range state.PlacedCanaries {
+				if alloc.ID == canary {
+					found = true
```

Comment:
You can break once you find the element
```suggestion
					found = true
					break
```

Reply:
:sweat_smile: woops thanks